common/djangoapps/student/tests/test_enrollment.py

```diff
@@ -55,6 +55,7 @@ def setUp(self):
         # We should NOT be auto-enrolled, because that would be giving
         # away an expensive course for free :)
         (['professional'], 'course_modes_choose', None),
+        (['no-id-professional'], 'course_modes_choose', None),
```

Comment:
There are a *lot* of test cases that use `ddt` to test different course modes.  Even if you have 100% diff coverage, it would be worth searching for these test cases and adding another line for the "no-id-professional" (like you did here).  This will help ensure that we don't accidentally break any of the many parts of the system that interact with course modes.